### ISSUE TITLE
Code not using GPU despite it being enabled

### ISSUE THREAD

**Issue by DPyrexx** [googlecolab/colabtools#4398]
I'm trying to use GPU to make the Gridsearch run faster but it runs it isn't using the GPU

`from google.colab import drive
drive.mount('/content/drive')

import tensorflow as tf
print("Tensorflow version " + tf.__version__)

# Check if GPU is available
if tf.test.gpu_device_name():
    print('Default GPU Device: {}'.format(tf.test.gpu_device_name()))
else:
    print("No GPU available. Please ensure you have configured GPU acceleration in your Colab runtime.")

from numpy import load
import numpy as np
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.metrics import classification_report, confusion_matrix
from sklearn.decomposition import PCA
from sklearn.neural_network import MLPClassifier
from sklearn.pipeline import Pipeline
import seaborn as sns
import matplotlib.pyplot as plt
import joblib

# Load data
path = '/content/drive/My Drive/'
y = load(path + 'yaleExtB_target.npy')
X = load(path + 'yaleExtB_data.npy')

# Split data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Define the parameter grid for grid search
param_grid = {
    'pca__n_components': [50, 100, 150, 200, None],
    'mlp__hidden_layer_sizes': [(50,), (100,), (200,)],
    'mlp__activation': ['tanh', 'relu'],
    'mlp__solver': ['sgd', 'adam'],
    'mlp__batch_size': [32, 64, 128, 256],
    'mlp__max_iter': [1000]
}

# Create the PCA object
pca = PCA(whiten=True)

# Create the MLP classifier
mlp = MLPClassifier()

# Create the pipeline
pipeline = Pipeline([
    ('pca', pca),
    ('mlp', mlp)
])

# Create the grid search object
grid_search = GridSearchCV(pipeline, param_grid, cv=3, scoring='accuracy', verbose=2)

# Fit the grid search to the data
grid_search.fit(X_train, y_train)

# Print the best hyperparameters
print("Best Hyperparameters:", grid_search.best_params_)

# Save the best model to disk
best_model = grid_search.best_estimator_
joblib.dump(best_model, 'best_model_with_pca.joblib')

# Evaluate the model with the best hyperparameters on the test set
y_pred = best_model.predict(X_test)

# Print classification report
print("Classification Report:")
print(classification_report(y_test, y_pred))`

**Comment by CL-VO** [comment]
more than 3 months of this issue and still nothing fixed.

Colab team showing how good they are once again...
